src/Elastic.Apm/Report/PayloadSenderV2.cs

```diff
@@ -91,6 +92,13 @@ internal class PayloadSenderV2 : IPayloadSender, IDisposable
 			string AdaptUserAgentValue(string value) => Regex.Replace(value, "[ /()<>@,:;={}?\\[\\]\"\\\\]", "_");
 		}
 
+		private static HttpClientHandler CreateHttpClientHandler(IConfigurationReader configurationReader) =>
+			new HttpClientHandler
+			{
+				ServerCertificateCustomValidationCallback = (message, certificate, chain, policyError) =>
+					policyError == SslPolicyErrors.None || !configurationReader.VerifyServerCert
```

Comment:
For supportability purposes I think it's worth to add logging (`DEBUG` level for example) if there is indeed an issue with the certificate that we are overriding and what is that issue (I assume it's in `policyError` parameter, right?)

Reply:
You'd want to log why certification validation fails, even when the user has specified to skip validation?

Reply:
I think it might be useful to know if the option has any effect and what can kind of validation issues does it override - we can lower the level to `TRACE` which is disabled by default.

Reply:
It's going to override all verification issues 😄 Logging can be added if you think it's worthwhile, but I'm not sure it is in this case, because setting `VerifyServerCert` to `false` says the user doesn't care.

Reply:
> It's going to override all verification issues

That is why logging the actual issue being overridden (`policyError` value) is useful.

> if you think it's worthwhile

I think it can helpful supportability feature. For example we might want to temporarily set `VerifyServerCert` to `false` just to see in logs what errors prevent normal HTTPS communication (I don't know if exception thrown on connection failure contains that level of detail).